File: csharp/armstrong-numbers/ArmstrongNumbers.cs
using System;
using System.Collections.Generic;
using System.Linq;

public static class ArmstrongNumbers
{
    public static int ToInt(char c) => (int)Char.GetNumericValue(c);

    public static bool IsArmstrongNumber(int number)
    {
        // getting rid of negatives
        if (number < 0)
        {
            return false;
        }
        // single digit numbers are armstrong numbers
        else if (number < 10)
        {
            return true;
        }
        // no double digit armstrong numbers exist
        else if (number < 100)
        {
            return false;
        }
        else
        {
            string num = number.ToString();
            int sum = ((int)(num.Select(ToInt)
                        .Select(b => Math.Pow(b, num.Length))
                        .Sum()));
            return sum == number;
        }
    }
}

File: dart/armstrong-numbers/lib/armstrong_numbers.dart
import 'dart:math';

class ArmstrongNumbers {
  bool isArmstrongNumber(int digit) {
    if (digit > 0 && digit < 10) {
      return true;
    } else if (9 < digit && digit < 100) {
      return false;
    } else {
      List<int> digits = [];
      digit
          .toString()
          .split('')
          .forEach((String d) => {digits.add(int.parse(d))});

      int sum = digits.fold(0, (int a, int b) {
        return a + pow(b, digits.length) as int;
      });
      return sum == digit;
    }
  }
}
